x-pack/platform/plugins/shared/fleet/server/services/agents/change_privilege_runner.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import type { ElasticsearchClient, SavedObjectsClientContract } from '@kbn/core/server';
+
+import { omit } from 'lodash';
+
+import pMap from 'p-map';
+
+import type { Agent } from '../../types';
+
+import { packagePolicyService } from '../package_policy';
+
+import { FleetUnauthorizedError } from '../../errors';
+
+import { MAX_CONCURRENT_AGENT_POLICIES_OPERATIONS } from '../../constants';
+
+import { ActionRunner } from './action_runner';
+import { BulkActionTaskType } from './bulk_action_types';
+
+import { createAgentAction, createErrorActionResults } from './actions';
+
+import { getPackagesWithRootAccess } from './change_privilege_level';
+
+export class ChangePrivilegeActionRunner extends ActionRunner {
+  protected async processAgents(agents: Agent[]): Promise<{ actionId: string }> {
+    return await BulkChangePrivilegeAgentsBatch(
+      this.esClient,
+      this.soClient,
+      agents,
+      this.actionParams!
+    );
+  }
+
+  protected getTaskType() {
+    return BulkActionTaskType.PRIVILEGE_LEVEL_CHANGE_RETRY;
+  }
+
+  protected getActionType() {
+    return 'PRIVILEGE_LEVEL_CHANGE';
+  }
+}
+
+export async function BulkChangePrivilegeAgentsBatch(
```

Comment:
the function should be lowercase for consistency
```suggestion
export async function bulkChangePrivilegeAgentsBatch(
```